### PR TITLE
forgot password backend functionality #12

### LINKED ISSUE #12: Implement Forgot Password Functionality
Develop the backend logic for the forgot password feature using the MERN stack. This functionality should enable users to reset their passwords by verifying their email address. Upon request, a secure link or OTP will be sent to the user's registered email, allowing them to reset their password safely.
Screenshot 2024-10-18 184318

please assign me in this issue and give me some day i can work on it

### PR DESCRIPTION
**Pull request by skanarul8002**
please accept the pr for issue #12, previous pr #19. 

**Title:** Implement Password Reset Functionality with Environment Configuration

**Description:**
    Fix #12 
- Added **password reset** functionality:
  - Users can request a reset link via `/resetPasswordLink`, and an email is sent using `nodemailer`.
  - Users reset their password via `/resetpassword/:id/:token`, verified using `jsonwebtoken`.
  - Integrated **`bcrypt`** for secure password hashing.
  - Handles expired tokens and invalid requests.

- Updated **.env** configuration for easy environment management:
  ```
  PORT=8081
  MONGODB_URI=mongodb://127.0.0.1:27017/your mongodb
  FRONTEND_URL=http://localhost:3000
  JWT_EXPIRY=1h
  JWT_SECRET=Skanarul$123@
  PASSWORD=mail permission password
  EMAIL=your email
  BACKEND_URL=http://localhost:8081/
  ```

- Enhanced **login/signup**:
  - Consistent JWT usage with expiration.
  - Secure password storage using `bcrypt`.

**Summary**: This update provides a robust password reset mechanism, improved security practices, and streamlined environment configuration for easier deployment.

See below video:
https://drive.google.com/file/d/1ZGETgRYIpFFSRAQ5gCf7U72PcEgfsm93/view?usp=drive_link

@Avdhesh-Varshney please merge the pr and fix the issue #12

### CODE REVIEW

File: middleware/fetchuser.js [hunk 1]
```diff
@@ -1,17 +1,22 @@
 import jwt from 'jsonwebtoken';
-const JWT_SECRET = process.env.JWT_SECRET;
+const JWT_SECRET = process.env.JWT_SECRET || 'Skanarul$123@';
```
Comment (Avdhesh-Varshney) [inline middleware/fetchuser.js:2]: Remove this update

**Review by Avdhesh-Varshney — CHANGES_REQUESTED** [review]

**Comment by skanarul8002** [comment]
@Avdhesh-Varshney , i have done your given changes request.

**Review by Avdhesh-Varshney — APPROVED** [review]

**Comment by skanarul8002** [comment]
@Avdhesh-Varshney , Please use labels gssoc-ext, levelno etc.

**Comment by Avdhesh-Varshney** [comment]
> @Avdhesh-Varshney , Please use labels gssoc-ext, levelno etc.

Backend repo is not part of `gssoc-ext`

**Comment by skanarul8002** [comment]
Good mentor @Avdhesh-Varshney , frontend is part of gssoc-ext but you are not accepted the pr code due to your own deployment code problem but I give more time to solve the assigned issue. Please give value to others contributor and getting value from others. As mentor, your behaviour is not good... Please make yourself as supportive person.

**Pull request merged**